runtime/tools/spec-builder/test/build-spec.spec.js

```diff
@@ -0,0 +1,37 @@
+const assert = require('assert');
+const fs = require('fs');
+const { buildSpec, emptyActualRawFiles } = require('./utils/spawn-child-process');
+const { isEqual } = require('lodash');
+
+
+before(async function() {
```

Comment:
1) It is better to actually build the spec files as part of the `it` tests, because the thing that you want to test should be executed in the test and not before the test 
2) it is better for the tests to be completely decoupled from one another. F.e if the first `buildSpec` fails, the other 2 will not be performed and all of the tests will fail.

The better approach is to:
1) Create one big describe f.e `Build-spec tests`
2) Put `before` function that clears the `actualRawFiles` directory
3) Create `it' tests that have the following logic:
       1 - build the spec json
       2- assert that the raw files are generated
       3- assert that the raw files are generated correctly
4) Put `after` function that clears the testbed (meaning deleting the `actualRawFiles` directory)